lib/CArtHandler.cpp

```diff
@@ -540,12 +532,12 @@ void CArtHandler::loadGrowingArt(CGrowingArtifact * art, const JsonNode & node)
 {
 	for (auto b : node["growing"]["bonusesPerLevel"].Vector())
 	{
-		art->bonusesPerLevel.push_back(std::pair <ui16, Bonus>((ui16)b["level"].Float(), Bonus()));
+		art->bonusesPerLevel.emplace_back(static_cast<ui16>(b["level"].Float()), Bonus());
 		JsonUtils::parseBonus(b["bonus"], &art->bonusesPerLevel.back().second);
 	}
 	for (auto b : node["growing"]["thresholdBonuses"].Vector())
 	{
-		art->thresholdBonuses.push_back(std::pair <ui16, Bonus>((ui16)b["level"].Float(), Bonus()));
+		art->thresholdBonuses.emplace_back(static_cast<ui16>(b["level"].Float()), Bonus());

```

Comment:
It is preffered to use emplace_back over push_back?

Reply:
This is a small optimization which removes a calling of copy constructor of something which we adding to vector) So, I think yes.